taskvine/src/manager/vine_file.c

```diff
@@ -147,27 +161,37 @@ void vine_file_delete(struct vine_file *f)
 
 struct vine_file * vine_file_local( const char *source )
 {
-	return vine_file_create(source,0,0,0,VINE_FILE,0,0);
+	return vine_file_create(source,0,0,0,0,VINE_FILE,0,0);
 }
 
 struct vine_file * vine_file_url( const char *source )
 {
-	return vine_file_create(source,0,0,0,VINE_URL,0,0);
+	return vine_file_create(source,0,0,0,0,VINE_URL,0,0);
+}
+
+struct vine_file * vine_file_substitute_url( struct vine_file *f, const char *source )
+{
+	return vine_file_create(source,0,f->cached_name,0,f->length,VINE_URL,0,0);
+}
+
+struct vine_file * vine_file_temp()
+{
+	return vine_file_create("temp",0,0,0,0,VINE_TEMP,0,0);
```

Comment:
As above, suggest adding a single argument as a cache name, or the basis of a cache name?